include/yarp/version.h

```diff
@@ -0,0 +1,5 @@
+#define YP_VERSION_MAJOR 0
+#define YP_VERSION_MINOR 4
+#define YP_VERSION_PATCH 0
+
+#define YP_VERSION "0.4.0"
```

Comment:
We should probably reuse the old logic to create this string from the components, so they can't get out of sync: https://github.com/ruby/yarp/pull/1088/files#diff-71ae829a627932e547cf825850d9a6b4773b009b0b68ae8eb1d8ef02e6d98987